react/react installer/e257d740-d47d-4794-82f1-3704850bdba9.json

```diff
@@ -0,0 +1,56 @@
+{
+  "implementations": {
+    "es6-node": {
+      "inputFunctions": {
+        "df3107d7-179c-4e60-86b4-323813c9e86e": "const { exec } = require( \"child_process\" );\nconst { projectPath } = data;\nconst { replace } = params;\nlet command = \"npx create-react-app \" + data + \" --use-npm\";\nif (state.running){\n  outputs.bounced(data, tag);\n} else {\n  state.running = true;\n  if ( params.replace === true ) {\n    command = \"rm -rf \" + data + \" && npx create-react-app \" + data + \" --use-npm\";\n  }\n  outputs.stdout( command, tag );\n  exec(command, (error, stdout, stderr) => {\nif (error && (params.replace === false || 'undefined' )) {\n  outputs.error(`Application folder already exists. Set the replace input to true if you wish to reinstall the application from scratch`, tag);\n  outputs.trigger(false, tag);\n} else {\n  outputs.stdout(stdout || \"\", tag);\n  outputs.stderr(stderr || \"\", tag);\n  outputs.error(null, tag);\n  outputs.trigger(true, tag);\n}\n  state.running = false;\n  });\n}"
```

Comment:
The statement `const { projectPath } = data;` is useless, as `data` is a string that contains the projectPath directly.

Reply:
I think it is way too dangerous to run a command like this without any validation.
```
command = \"rm -rf \" + data + \" && npx create-react-app \" + data + \" --use-npm\";
```
I think we'd need to be sure the folder specified in `data` is really a folder and it really contains a react-app from a previous run.

Reply:
`outputs.error(null, tag);`
No signal should be sent to an error port if there were no errors.
On the other hands if there were any the sent out value should be wrapped to this type at minimum: `{"error": string}`

Reply:
> The statement `const { projectPath } = data;` is useless, as `data` is a string that contains the projectPath directly.

agree - const removed.

Reply:
> I think it is way too dangerous to run a command like this without any validation.
> 
> ```
> command = \"rm -rf \" + data + \" && npx create-react-app \" + data + \" --use-npm\";
> ```
> 
> I think we'd need to be sure the folder specified in `data` is really a folder and it really contains a react-app from a previous run.

agree - when the node creates a react app now it is additionally adding a file .cranq-react-app to make sure that is I want to delete the node before restarting it can check for the file as well as the path before deciding to remove. As an additional I realised the rm -rf was not of much use to win32 users so I have used fs.rmSync so that all os platforms will behave the same.

Reply:
> `outputs.error(null, tag);` No signal should be sent to an error port if there were no errors. On the other hands if there were any the sent out value should be wrapped to this type at minimum: `{"error": string}`

this is reused code from another node in the repository. I have now deleted the line. Additionally I have taken note of the errors being in typed to { "error" : "" } and have amended as necessary.